src/Link/LinkProcessor.php

```diff
@@ -35,7 +34,30 @@ public static function preRenderLink($element) : array {
           $element['#attributes']['data-protocol'] = $scheme;
         }
       }
-      $element['#title']['#attributes'] = $element['#attributes'] ?? [];
+
+      if (!empty($element['#attributes']['data-is-external'])) {
+        $use_helfi_link = TRUE;
+      }
+      if (!empty($element['#attributes']['data-protocol'])) {
+        $use_helfi_link = TRUE;
+      }
+
+      /** @var \Drupal\Core\Url $url */
+      $url = $element['#url'];
+      $url_attributes = $url->getOption('attributes');
+      if (!empty($url_attributes['data-selected-icon'])
```

Comment:
Is this condition necessary? I think it's set by hdbt in https://github.com/City-of-Helsinki/drupal-hdbt/blob/main/templates/module/helfi_api_base/helfi-link.html.twig#L4 and the template is only used when the link is rendered through `helfi_link` template